File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0;
        int l = 0;
        for(int r=0;r<prices.size();r++) {
            while(prices[l] > prices[r] && l < r) {
                l++;
            }
            int d = prices[r] - prices[l];
            profit = max(profit, d);
        }
        return profit;
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n = len(prices)
        l,r = 0,0
        res = 0
        while r < n:
            if prices[l] < prices[r]:
                profit = prices[r] - prices[l]
                res = max(res, profit)
            else:
                l = r
            r += 1
        return res